modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -1539,6 +1542,85 @@ public void deploysAppZipWithExtensionXmlPlugin() throws Exception {
+  }
+
+  @Test
+  public void deploysWithExtensionXmlPluginWithPomWithMulePluginDependenciesThatArePresentInTheClasspath() throws Exception {
+    String moduleFileName = "module-bye.xml";
+    String extensionName = "bye-extension";
+
+    MulePluginModelBuilder builder =
+        new MulePluginModelBuilder().setName(extensionName).setMinMuleVersion("4.0.0");
+    builder.withExtensionModelDescriber().setId(XmlExtensionModelLoader.DESCRIBER_ID).addProperty(RESOURCE_XML, moduleFileName);
+    builder.withClassLoaderModelDescriber().setId(MAVEN_ID_CLASSLOADER_MODEL_DESCRIPTOR)
+        .addProperty(ArtifactPluginDescriptorFactory.EXPORTED_PACKAGES, Arrays.asList("org.foo"));
+
+    final ArtifactPluginFileBuilder byeXmlExtensionPlugin = new ArtifactPluginFileBuilder(extensionName)
+        .containingResource("module-byeSource.xml", moduleFileName)
+        .containingMetaInfResource("module-bye-pom-with-mule-plugin-dependencies.xml", "pom.xml")
+        .describedBy(builder.build());
+
+    ApplicationFileBuilder applicationFileBuilder = new ApplicationFileBuilder("appWithExtensionXmlPlugin")
+        .definedBy("app-with-extension-xml-plugin-config.xml").containingPlugin(byeXmlExtensionPlugin)
+        .containingPlugin(echoPlugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+
+    assertDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+  }
+
+  @Test
+  public void failsToDeployWithExtensionXmlPluginWithPomWithMulePluginDependenciesThatAreAbsentInTheClasspath() throws Exception {
+    String moduleFileName = "module-bye.xml";
+    String extensionName = "bye-extension";
+
+    MulePluginModelBuilder builder =
+        new MulePluginModelBuilder().setName(extensionName).setMinMuleVersion("4.0.0");
+    builder.withExtensionModelDescriber().setId(XmlExtensionModelLoader.DESCRIBER_ID).addProperty(RESOURCE_XML, moduleFileName);
+    builder.withClassLoaderModelDescriber().setId(MAVEN_ID_CLASSLOADER_MODEL_DESCRIPTOR)
+        .addProperty(ArtifactPluginDescriptorFactory.EXPORTED_PACKAGES, Arrays.asList("org.foo"));
+
+    final ArtifactPluginFileBuilder byeXmlExtensionPlugin = new ArtifactPluginFileBuilder(extensionName)
+        .containingResource("module-byeSource.xml", moduleFileName)
+        .containingMetaInfResource("module-bye-pom-with-mule-plugin-dependencies.xml", "pom.xml")
+        .describedBy(builder.build());
+
+    ApplicationFileBuilder applicationFileBuilder = new ApplicationFileBuilder("appWithExtensionXmlPlugin")
+        .definedBy("app-with-extension-xml-plugin-config.xml").containingPlugin(byeXmlExtensionPlugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+
+    // TODO(fernandezlautaro): MULE-11089 the following line is expecting to see two deploys when it MUST be just one, for some reason tries to redeploy two times so I added until this bug gets fixed
```

Comment:
the following line is expecting to see two deploys when it MUST be just one >the following line is expecting to see one deploy